src/sql/ConfigEditor/ConfigEditor.tsx

```diff
@@ -0,0 +1,63 @@
+import React, { useState } from 'react';
+import { DataSourcePluginOptionsEditorProps, DataSourceSettings } from '@grafana/data';
+import { AwsAuthDataSourceJsonData, AwsAuthDataSourceSecureJsonData, AwsAuthDataSourceSettings } from '../../types';
+import { getBackendSrv } from '@grafana/runtime';
+import { Selector, SelectorInput } from './Selector';
+import { InlineText, TextInput } from './InlineText';
+import { ConnectionConfig } from 'ConnectionConfig';
+
+type SQLDataSourceJsonData<T extends AwsAuthDataSourceJsonData> = T & {
+  [n: string]: any;
+};
+
+export interface SQLConfigEditorProps
+  extends DataSourcePluginOptionsEditorProps<SQLDataSourceJsonData<{}>, AwsAuthDataSourceSecureJsonData> {
+  inputs: Array<SelectorInput | TextInput | JSX.Element>;
```

Comment:
I wonder if it would be simpler to use props.children and always pass existing props and saveOptions  to those children and let the consumer of SQLConfigEditor have control over what goes inside it and whether or not it uses saveOptions. Might make it more flexible for the future as well? But maybe too flexible? Idk. Just sharing a thought in case it's helpful

Reply:
cool seems sensible to me! If you find yourself not wanting saveOptions duplicated this package could export a hook useSaveOptions or something like that but also it makes sense to me that the individual packages would have that logic.